lib/Dialect/SV/Transforms/HWMemSimImpl.cpp

```diff
@@ -260,8 +260,41 @@ void HWMemSimImpl::generateMemory(HWModuleOp op, FirMemory mem) {
     // Read Logic
     Value rdata = getMemoryRead(b, reg, addr, addMuxPragmas);
     if (!ignoreReadEnableMem) {
-      Value x = b.create<sv::ConstantXOp>(rdata.getType());
-      rdata = b.create<comb::MuxOp>(en, rdata, x, false);
+      size_t addrRange = 1 << addr.getType().getIntOrFloatBitWidth();
```

Comment:
```suggestion
      size_t addrRange = 1ULL << addr.getType().getIntOrFloatBitWidth();
```
or maybe `size_t{1}` or something?

Concern: While result is `size_t`, the expression is evaluated in terms of integer promotion, here looks like signed int.

As-is, this will produce wrong value (and undefined behavior) when returned width is >= 31 (shifting /into/ sign bit is not UB in /all/ C++ language versions but is in latest IIRC-- anyway, 32 sure is).

Here's a little reproducer to demonstrate:

```c++
#include <cstdio>

size_t foo(unsigned x) {
  size_t result = 1 << x;
  return result;
}

int main() {
  printf("%zu\n", foo(0));
  printf("%zu\n", foo(31));
  printf("%zu\n", foo(32));
  printf("%zu\n", foo(33));

  return 0;
  ```
  
 Building w/ `-fsanitize=integer` will identify and report the truncation.

Reply:
In C++23 this could be `1UZ` but no size_t literals presently, apparently/FWIW: https://en.cppreference.com/w/cpp/language/integer_literal .

Reply:
Thanks Will for catching this. Fixed it.